internal/docs/design.md

```diff
@@ -204,6 +204,27 @@ needed). However, this would result in inconsistent names over time (e.g., some
 packages would expose `New` with an `Options`, while others would expose
 `NewWithOptions`).
 
+### Compound IDs
+
+Many cloud providers have resource IDs that are made up of subcomponents in
+some well-defined syntax. For example, [GCP KMS key IDs][] take the form
+`projects/[PROJECT_ID]/locations/[LOCATION]/keyRings/[KEY_RING]/cryptoKeys/[KEY]`.
+We call these _compound IDs_.
+
+There are two broad compound ID usage patterns we have observed:
+
+1. Applications will keep resources in the same location or project, so the
+   application will build the ID from the subcomponents rather than passing the
+   entire resource ID around.
+2. Applications will pass a verbatim string from configuration down to the
+   API, since this is what was easily copy-pasteable from the cloud console UI.
+
+Go CDK constructors that take in compound IDs should take in a `string` with the
```

Comment:
Nit: add "and URLOpeners".

Reply:
But URLOpeners take in a `*url.URL`, not a `string`. Can you elaborate on the change you want to see here?

Reply:
I mean that the URL used to open the resource should have the full compound ID in them. So, in your example, the URL should have `projects/[PROJECT_ID]/locations/[LOCATION]/keyRings/[KEY_RING]/cryptoKeys/[KEY]` in it somewhere.

Reply:
Gotcha. Added a slightly different wording to be more precise, PTAL.

Reply:
SGTM.